API/app/Http/Controllers/CdsController.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Cds;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use App\Http\Requests\RetrieveOpis;
+
+class CdsController extends Controller
+{
+    /**
+     * Ottieni una lista di tutti i cds relativi
+     * all'ultimo anno accademico (scelta effettuata 
+     * nella versione precedente). 
+     */
+    public function index () 
+    {
+        $newerCds = Cds::getNewer();     
+
+        return response()->json($newerCds, Response::HTTP_OK); 
+    }
+
+    /**
+     * Ottieni tutti gli insegnamenti di un CDS relativi
+     * all'ultimo anno accademico (scelta effettuata 
+     * nella versione precedente). 
+     */
+    public function getTeachings (int $cds)
+    {
+        $cds = Cds::find($cds); 
+
+        if (null === $cds) 
```

Comment:
logicamente controlli se null e' uguale a $cds, io lo scriverei al contrario per questioni di readability e "logiche", io voglio controllare se cds e' null, non se null e' uguale a cds.

ma per fare qualche controllo in piu' con meno codice, ti direi di utilizzare:
```php
if (!!$cds) {
   ...
}
```

questo doppio "!!" converte la tua variabile in un valore booleano e di fatto, "under the hood" controlla se la tua variabile non e' `null, stringa vuota ('') oppure 0`

Reply:
Si ma è più espressivo null === $cds, si capisce subito l'intento

Reply:
Ma fai un solo check, in quel modo ne fai 3 (almeno).